cmd/dp-hierarchy-api/main.go

```diff
@@ -86,24 +86,32 @@ func main() {
 
 	// gracefully shutdown the application, closing any open resources
 	logData["timeout"] = config.ShutdownTimeout
-	log.Event(ctx, "start shutdown", log.ERROR, log.Error(err), logData)
+	log.Event(ctx, "start shutdown", log.INFO, logData)
 	shutdownContext, shutdownContextCancel := context.WithTimeout(context.Background(), config.ShutdownTimeout)
-
-	hc.Stop()
+	hasShutdownError := false
 
 	go func() {
+
+		log.Event(ctx, "stopping health checks", log.INFO)
```

Comment:
I dont think we need a log event before and after each step, surely if you just logged before, the fact that we get to the next log line is proof enough that the previous step succeeded? and then if it was hanging for any reason we'd see 'stopping X' as the last line and know thats where it got stuck?

Means you can simplify the logic below, and not have to have the else cases too